tools/rpkg/tests/testthat/test_extensions.R

```diff
@@ -0,0 +1,37 @@
+library("DBI")
+library("testthat")
+
+test_that("extensions can be loaded", {
+  con <- dbConnect(duckdb::duckdb())
+  on.exit(dbDisconnect(con, shutdown = TRUE))
+  matches = list.files(pattern='*.duckdb_extension', recursive=FALSE, full.names=TRUE)
+  matches <- c(matches, list.files(path='/tmp/duckdb_extensions', pattern='*.duckdb_extension', recursive=TRUE, full.names=TRUE))
+
+  require_variable <- Sys.getenv("DUCKDB_R_TEST_EXTENSION_REQUIRED")
+
+  if (require_variable == "1") {
+    expect_true(length(matches) > 0, "DUCKDB_R_TEST_EXTENSION_REQUIRED was set, but no extensions were found")
+  } else {
+    if (length(matches) == 0) {
+        warning("No extensions found for testing!")
+    }
+  }
+
+  for (extension in matches){
+    # Test loading the extension
+    query <- paste("LOAD '", extension, "';", sep = "")
+    dbExecute(con, query)
```

Comment:
Should here also be asome `expect_` function to actually test if the LOAD was successful?

Reply:
The thing is that the load statement does not return anything. If the dlopen call in the load statement fails, it will just crash so I think that just running the query, combined with the simple functional test for the httpfs extension, should be fine for now